rules/geneticrules.js

```diff
@@ -9,353 +9,370 @@ var MUT_COLOR_TRUNC = '#000000';
-	'disp_mut': {
-	    'missense_rec,inframe_rec,trunc_rec': {
-		shapes: [{
-			'type': 'rectangle',
-			'fill': MUT_COLOR_MISSENSE,
-			'x': '0%',
-			'y': '33.33%',
-			'width': '100%',
-			'height': '33.33%',
-			'z': 6
-		}],
-		legend_label: 'Mutation (putative driver)'
-	    },
-	    'missense,inframe,trunc,promoter,promoter_rec': { 
-		shapes: [{
-			'type': 'rectangle',
-			'fill': MUT_COLOR_MISSENSE_PASSENGER,
-			'x': '0%',
-			'y': '33.33%',
-			'width': '100%',
-			'height': '33.33%',
-			'z': 6
-		}],
-		legend_label: 'Mutation (putative passenger)'
-	    },
-	},
+        'disp_mut': {
+            'missense_rec,inframe_rec,trunc_rec,missense_rec_germ,inframe_rec_germ,trunc_rec_germ': {
+                shapes: [{
+                    'type': 'rectangle',
+                    'fill': MUT_COLOR_MISSENSE,
+                    'x': '0%',
+                    'y': '33.33%',
+                    'width': '100%',
+                    'height': '33.33%',
+                    'z': 6
+                }],
+                legend_label: 'Mutation (putative driver)'
+            },
+            'missense,inframe,trunc,promoter,promoter_rec,missense_germ,inframe_germ,trunc_germ,promoter_germ,promoter_rec_germ': {
+                shapes: [{
+                    'type': 'rectangle',
+                    'fill': MUT_COLOR_MISSENSE_PASSENGER,
+                    'x': '0%',
+                    'y': '33.33%',
+                    'width': '100%',
+                    'height': '33.33%',
+                    'z': 6
+                }],
+                legend_label: 'Mutation (putative passenger)'
+            },
+        },
     })
 };
 window.geneticrules.genetic_rule_set_different_colors_no_recurrence = {
-    'type':'gene',
+    'type': 'gene',
     'legend_label': 'Genetic Alteration',
     'rule_params': $.extend({}, non_mutation_rule_params, {
-	'disp_mut': {
-	    'promoter,promoter_rec': {
-		shapes: [{
-			'type': 'rectangle',
-			'fill': MUT_COLOR_PROMOTER,
-			'x': '0%',
-			'y': '33.33%',
-			'width': '100%',
-			'height': '33.33%',
-			'z': 6,
-		    }],
-		legend_label: 'Promoter Mutation'
-	    },
-	    'trunc,trunc_rec': {
-		shapes: [{
-			'type': 'rectangle',
-			'fill': MUT_COLOR_TRUNC,
-			'x': '0%',
-			'y': '33.33%',
-			'width': '100%',
-			'height': '33.33%',
-			'z': 6,
-		    }],
-		legend_label: 'Truncating Mutation',
-	    },
-	    'inframe,inframe_rec': {
-		shapes: [{
-			'type': 'rectangle',
-			'fill': MUT_COLOR_INFRAME,
-			'x': '0%',
-			'y': '33.33%',
-			'width': '100%',
-			'height': '33.33%',
-			'z': 6,
-		    }],
-		legend_label: 'Inframe Mutation',
-	    },
-	    'missense,missense_rec': {
-		shapes: [{
-			'type': 'rectangle',
-			'fill': MUT_COLOR_MISSENSE,
-			'x': '0%',
-			'y': '33.33%',
-			'width': '100%',
-			'height': '33.33%',
-			'z': 6,
-		    }],
-		legend_label: 'Missense Mutation',
-	    },
-	}
+        'disp_mut': {
+            'promoter,promoter_rec,promoter_germ,promoter_rec_germ': {
```

Comment:
confused - so you both denote germline with the `disp_mut` value and the `disp_germline` value? why is that necessary?

Reply:
This to make a room of a possibility where germline mutations (all with '_germ' suffix) are displayed in different color or  different order.

Reply:
I see, that make sense. I'm concerned about how complicated this makes it now, with germ, rec, rec_germ. Maybe as long as we don't need separate graphics, we can just omit this - make disp_germ render as the small white line, and we can handle the different order inside the comparator function, which would have access to disp_mut and disp_germ

Reply:
I can't think of any better solution for this for now especially for dealing with the ordering and sorting in the cbioportal-frontend part

Reply:
what do you mean? we can add logic to check for disp_germline around oncoprint/SortUtils.ts:62

Reply:
There is also prioritization logic. https://github.com/thehyve/cbioportal-frontend/blob/3ab38b7188daab3034ef3efe047224cfd0c2775d/src/shared/components/oncoprint/DataUtils.ts#L33
Without the suffix, it's more challenging to prioritize the rendering.

Reply:
I just really don't like redundancy.. it makes maintenance more confusing and error-prone. In this case I think we could replace the `priority` map argument with a function that maps a datum to its order value, and that function will make use of the `disp_mut` value map, and the `disp_germline` value.

Reply:
@adamabeshouse is that the only thing blocking the merging of this PR and PR #913?

Reply:
@Sjoerd-van-Hagen yes I suppose so

Reply:
@adamabeshouse okay so I've changed disp_germline to isGermline and this flag is triggered by https://github.com/cBioPortal/cbioportal-frontend/pull/913/files#diff-bf5c432c23e8bb3755ca9a6b9da2af78R168